lib/usd/ui/CMakeLists.txt

```diff
@@ -51,6 +51,10 @@ target_compile_definitions(${PROJECT_NAME}
         MAYAUSD_UI_EXPORT
 )
 
+# QT_NO_KEYWORDS prevents Qt from defining the foreach, signals, slots and emit macros.
+# this avoids overlap between Qt macros and boost, and enforces using Q_ macros.
+set_target_properties(Qt5::Core PROPERTIES INTERFACE_COMPILE_DEFINITIONS QT_NO_KEYWORDS)
+
```

Comment:
It's not obvious to me how these Qt changes are related to Python 3 support.

Reply:
In Python 3 headers, there are certain identifiers such as 'signals' and/or 'slots' that conflict with Qt's identifiers. To get around this one needs to replace all special Qt keywords ( signals, slots, emit ) with their Q_ macro counterparts. 

This is a known issue. Please see Qt's help ( Using Qt with 3rd Party Signals and Slots )

https://doc.qt.io/qt-5/signalsandslots.html#using-qt-with-3rd-party-signals-and-slots

Here is the error that one gets before the fix:

C:\Python37\include\object.h(448): error C2059: syntax error: ';'
error C2238: unexpected token(s) preceding ';'

and here is the line in question is PyType_Slot *slots; /* terminated by slot==0. */
```
typedef struct{
    const char* name;
    int basicsize;
    int itemsize;
    unsigned int flags;
    PyType_Slot *slots; /* terminated by slot==0. */
} PyType_Spec;
```